io/shared/src/main/scala/fs2/io/net/Socket.scala

```diff
@@ -69,6 +70,65 @@ trait Socket[F[_]] {
   /** Writes the supplied stream of bytes to this socket via `write` semantics.
     */
   def writes: Pipe[F, Byte, Nothing]
+
+  /** Reads a file in chunks and writes it to a socket.
+    * Streams the file contents in chunks of the specified size and sends them over the socket.
+    * The stream terminates when the entire file has reached end of file or the specified count is reached.
+    *
+    * @param file the file handle to read from
+    * @param offset the starting position in the file
+    * @param count the maximum number of bytes to transfer
+    * @param chunkSize the size of each chunk to read
+    */
+  def sendfile(
+      file: FileHandle[F],
+      offset: Long,
+      count: Long,
+      chunkSize: Int
+  ): Stream[F, Unit] = {
```

Comment:
`Stream[F, Unit]` is a stream of 0 to infinite `()`s. `Stream[F, Nothing]` is definitively an empty stream.

```suggestion
  ): Stream[F, Nothing] = {
```